File: commons/src/main/kotlin/io/lucidtech/commons/async/FutureWrapper.java
package io.lucidtech.commons.async;

import java.util.concurrent.Future;
import java.util.concurrent.TimeUnit;

public class FutureWrapper<T> implements Future<T> {
    private final T result;
    private boolean isCancelled = false;

    public FutureWrapper(final T result) {
        this.result = result;
    }

    @Override
    public boolean cancel(boolean mayInterruptIfRunning) {
        isCancelled = true;
        return true;
    }

    @Override
    public boolean isCancelled() {
        return isCancelled;
    }

    @Override
    public boolean isDone() {
        return true;
    }

    @Override
    public T get() {
        return result;
    }

    @Override
    public T get(long timeout, TimeUnit unit) {
        return result;
    }
}

File: commons/src/main/kotlin/io/lucidtech/commons/async/FutureWrapper.kt
package io.lucidtech.commons.async

import java.util.concurrent.Future
import java.util.concurrent.TimeUnit

class FutureWrapper<T>(private val result: T) : Future<T> {
    private var isCancelled = false

    override fun cancel(mayInterruptIfRunning: Boolean): Boolean {
        isCancelled = true
        return true
    }

    override fun isCancelled(): Boolean = isCancelled
    override fun isDone(): Boolean = true
    override fun get(): T = result
    override fun get(timeout: Long, unit: TimeUnit): T = result
}
